src/file-uploader/file-uploader.component.ts

```diff
@@ -241,35 +242,37 @@ export class FileUploader implements ControlValueAccessor {
 		event.stopPropagation();
 		event.preventDefault();
 
-		const transferredFiles = Array.from(event.dataTransfer.files);
+		const transferredFiles: Array<File> = Array.from(event.dataTransfer.files);
+		const newFiles = new Set<FileItem>(this.files);
 
 		transferredFiles.filter(({ name, type }) => {
 			// Get the file extension and add a "." to the beginning.
 			const fileExtension = name.split(".").pop().replace(/^/, ".");
 			// Check if the accept array contains the mime type or extension of the file.
 			return this.accept.includes(type) || this.accept.includes(fileExtension) || !this.accept.length;
 		}).forEach(file => {
-			if (!this.files.size || this.multiple) {
+			if (!newFiles.size || this.multiple) {
 				const fileItem = this.createFileItem(file);
-				this.files.add(fileItem);
+				newFiles.add(fileItem);
 			}
 		});
 
-		this.filesChange.emit(this.files);
-		this.value = this.files;
+		this.value = newFiles;
+		this.filesChange.emit(newFiles);
 		this.dragOver = false;
 	}
 
 	removeFile(fileItem) {
-		let shouldEmit = true;
-		if (this.files) {
-			shouldEmit = this.files.has(fileItem);
-			this.files.delete(fileItem);
```

Comment:
If `this.files.has(fileItem)` is `false`, then `this.files.delete(fileItem);` doesn't do anything, so it's redundant.